cloudprovider/alibabacloud/slb.go

```diff
@@ -144,8 +144,11 @@ func initLbCache(svcList []corev1.Service, minPort, maxPort int32, blockPorts []
 			var ports []int32
 			for _, port := range getPorts(svc.Spec.Ports) {
 				if port <= maxPort && port >= minPort {
-					newCache[lbId][port] = true
-					ports = append(ports, port)
+					value, ok := newCache[lbId][port]
```

Comment:
you cannot redeploy the service that not controlled by kruise-game but use same port.

Reply:
cache just save lbId + port when design,  it has no Protocol Attribute；

when use 8000/TCPUDP,  cache just lbid:8000  and map to tcp:8000 udp:8000  two svc items;

when reinit cache , list tcp:8000 and udp:8000 should repeat to lbid:8000 

else when 8000/TCPUDP, 9000/TCP will map error ....

Reply:
here svc all filter by SlbIdLabelKey , all  controlled by kruise-game ?

Reply:
I don't think there is a problem to set map true for two times. And the svc using SlbIdLabelKey may be deployed by user self, not all controlled by kruise-game.

Reply:
1.  set map true two times has no problem ,  but append two times maybe problem;   eg:   when config 8000/TCPUDP,9000/TCP ,  first create cache is lb:8000,9000 ,  when reinit cache with append two times, result is lb:8000,8000,9000,  it will be error when resync svc;

2. I will sort svc ports to ensure same port different protocols order,  when User define yaml diff order ....

Reply:
I got your point